python-frontend/src/main/java/org/sonar/python/parser/PythonParser.java

```diff
@@ -69,18 +76,19 @@ private static class SslrPythonParser extends Parser<Grammar> {
     private final LexerState lexerState;
     private final Lexer lexer;
 
-    private SslrPythonParser() {
-      super(PythonGrammar.create());
+    private SslrPythonParser(Grammar grammar, LexerState lexerState, Lexer lexer) {
+      super(grammar);
       super.setRootRule(super.getGrammar().getRootRule());
-      this.lexerState = new LexerState();
-      this.lexer = PythonLexer.create(lexerState);
+      this.lexerState = lexerState;
+      this.lexer = lexer;
     }
 
     @Override
     public AstNode parse(String source) {
       lexerState.reset();
       lexer.lex(source);
-      return super.parse(tokens());
+      List<Token> tokens = tokens();
+      return super.parse(tokens);
```

Comment:
why this change?

Reply:
this change was done to make it easier to debug parsing, but could be removed for sure if you think is required

Reply:
(We chose to leave this unchanged)